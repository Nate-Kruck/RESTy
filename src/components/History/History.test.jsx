import React from 'react';
import {
  render,
  cleanup
} from '@testing-library/react';
import History from './History';

describe('History Testing', () => {
  afterEach(() => cleanup());

  it('should render request from history', () => {
    const history = [
      {
        url: 'www.friend.com',
        method: 'GET'
      },
      {
        url: 'www.friend.com',
        method: 'POST'
      },
      {
        url: 'www.friend.com',
        method: 'DELETE'
      }
    ];
    
    const { asFragment } = render(<History
      history={history}
      url="blank"
      method="GET"
      onClick={() => {}}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
