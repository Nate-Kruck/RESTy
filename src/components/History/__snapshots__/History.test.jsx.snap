// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Testing should render request from history 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <p>
        GET
      </p>
      <p>
        www.friend.com
      </p>
    </li>
    <li>
      <p>
        POST
      </p>
      <p>
        www.friend.com
      </p>
    </li>
    <li>
      <p>
        DELETE
      </p>
      <p>
        www.friend.com
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
